internal/command/binding_list.go

```diff
@@ -0,0 +1,180 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package command
+
+import (
+	"fmt"
+
+	camelkv1alpha1 "github.com/apache/camel-k/pkg/apis/camel/v1alpha1"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1beta1 "k8s.io/apimachinery/pkg/apis/meta/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/client/pkg/kn/commands"
+	"knative.dev/client/pkg/kn/commands/flags"
+	hprinters "knative.dev/client/pkg/printers"
+)
+
+var bindingListExample = `
+  # List Kamelet bindings.
+  kn-source-kamelet binding list
+
+  # List available Kamelet bindings in YAML output format
+  kn-source-kamelet binding list -o yaml`
+
+// newBindingCreateCommand implements 'kn-source-kamelet binding list' command
+func newBindingListCommand(p *KameletPluginParams) *cobra.Command {
+	listFlags := flags.NewListPrintFlags(ListBindingHandlers)
+
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List Kamelet bindings.",
+		Aliases: []string{"ls"},
+		Example: bindingListExample,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			namespace, err := p.GetNamespace(cmd)
+			if err != nil {
+				return err
+			}
+
+			kameletClient, err := p.NewKameletClient()
+			if err != nil {
+				return err
+			}
+
+			bindingList, err := kameletClient.KameletBindings(namespace).List(p.Context, v1.ListOptions{})
+			if err != nil {
+				return err
+			}
+			if len(bindingList.Items) == 0 {
+				fmt.Fprintf(cmd.OutOrStdout(), "No resources found.\n")
+				return nil
+			}
+
+			// empty namespace indicates all-namespaces flag is specified
+			if namespace == "" {
+				listFlags.EnsureWithNamespace()
+			}
+
+			err = listFlags.Print(bindingList, cmd.OutOrStdout())
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+	commands.AddNamespaceFlags(cmd.Flags(), true)
+	listFlags.AddFlags(cmd)
+	return cmd
+}
+
+// ListBindingHandlers handles printing human readable table for `kn-source-kamelet binding list` command's output
+func ListBindingHandlers(h hprinters.PrintHandler) {
+	columnDefinitions := []metav1beta1.TableColumnDefinition{
+		{Name: "Namespace", Type: "string", Description: "Namespace of the Kamelet binding", Priority: 0},
+		{Name: "Name", Type: "string", Description: "Name of the Kamelet binding ", Priority: 1},
+		{Name: "Phase", Type: "string", Description: "Phase of the Kamelet binding ", Priority: 1},
+		{Name: "Age", Type: "string", Description: "Age of the Kamelet binding ", Priority: 1},
+		{Name: "Conditions", Type: "string", Description: "Ready state conditions", Priority: 1},
+		{Name: "Ready", Type: "string", Description: "Ready state of the Kamelet binding ", Priority: 1},
+		{Name: "Reason", Type: "string", Description: "Reason if state is not Ready", Priority: 1},
```

Comment:
Haven't check yet, but isn't this a bit much for printing out in default mode ? I wonder how it looks like on a 120 column wide terminal.

Mabye one could hide some of the not so needed options behind a `--verbose` ?


Also, the namespace should be only shown when used with an option `--all-namespaces` (or `-A`) which would make sense as an option, as well.

Reply:
well, just saw that `kn service list has even more columns`, so guess it's ok :)

But can you check that the Namespace column is only printed when --namespace is given ?

Reply:
the namespace is only printed when `--all-namespaces` is specified